docs/memory.md

```diff
@@ -0,0 +1,104 @@
+every 500 ms.  This helps to give a live appearance on the state of the memory as the application is running.  The quantities plotted on the
+y-axis are (from top to bottom):
+- Capacity - Current capacity of the heap.
+- GC - GC has occurred.
+- Used - Objects (Dart objects) in the the heap.
+- External - memory not in the Dart heap, but is retained (memory read from a file) or decoded image in Flutter.
+
+<img src="images/memory_rss_chart.png" width="800" />
+
+To view RSS click on the RSS name in the legend.
+
+- RSS is the Resident Set Size and is used to show how much memory is allocated to that process and is in RAM. It does not include memory that is swapped out. It does include memory from shared libraries as long as the pages from those libraries are actually in memory. It does include all stack and heap memory.
+
+For more detailed information on the [Dart VM](https://mrale.ph/dartvm/) and memory.
+
+## Event Timeline
+
+<img src="images/memory_parts.png" width="800" />
+
+Chart displaying when a DevTool event (Snapshot and Reset button clicked) in relation to the timeline of the memory chart.  Hovering over the markers in the Event timeline will display the time when the Snapshot or Reset happened.  This helps to correlate to when in the timeline (x-axis) a memory leak might have occurred.
+
+Clicking on the Snapshot button will show the current state of the memory (heap) with regards to all active classes their instances.  When the Reset button is pressed the accumulator for all classes reset to zero, notice that the reset is tied with a faint blue when a Snapshot is again clicked displays the new accumulators values since the last Reset. 
+
+
+## Snapshot Classes
+- Size - Total amount of memory used by current objects in the heap.
+- Count - Total number of current objects in the heap.
+- Accumulator - Total number of objects in the heap since the last reset.
+- Class - An aggregate of the objects allocated to this class. Clicking the class takes you to the list of all instances of this class.
+
+## Instances of a Class
+Displays a list of all instances by their handle name. _TODO INSPECTING AN INSTANCE_.
+
+## Memory Actions
+
+### Liveness of the Memory Usage Overview Chart
+- Pause - Pause the memory usage overview chart to allow inspecting of of the data currently being plotted.  New memory data continues to receive new data, notice the Range Selector grows, to the right.
+- Resume - The memroy usage overview chart is live and displaying the current time and the latest memory data received.
+
+### Managing the Objects and Statistics in the Heap
+- Snapshot - Returns the list of all active classes in the heap.  The Accumulator column displays the number of allocated objects since the previous "Reset".
+- Reset - Zeroes out the Accumulator column in the Snapshot Classes table and refreshes the displayed data.
+- Filter - TODO
```

Comment:
address this TODO

Reply:
- Filtering Classes - **_TODO: Document Filter dialog_**
- Inspecting contents of an instance - **_TODO: Document inspecting data values of an instance._**
Displays a list of all instances by their handle name. **_TODO: Add link to inspecting data values_**.